bigtable/internal/testproxy/proxy.go

```diff
@@ -789,7 +789,58 @@ func (s *goTestProxyServer) BulkMutateRows(ctx context.Context, req *pb.MutateRo
 // CheckAndMutateRow responds to the CheckAndMutateRow RPC. This method applies
 // one mutation if a condition is true and another mutation if it is false.
 func (s *goTestProxyServer) CheckAndMutateRow(ctx context.Context, req *pb.CheckAndMutateRowRequest) (*pb.CheckAndMutateRowResult, error) {
-	return nil, stat.Error(codes.Unimplemented, "CheckAndMutateRow not implemented")
+	s.clientsLock.Lock()
```

Comment:
RLock() and RUnlock() here.

Reply:
(Also it seems we've been missing this in the other methods)

Reply:
Done.